### ISSUE TITLE
Include categories in SliverAppBar

### ISSUE THREAD

**Issue by Docmale** [digitalfabrik/entitlementcard#1384]
**Is your feature request related to a problem? Please describe.**
There are around 5800 acceptance points in Bavaria, when I scroll down the list, the categories disappear and I have to move back up all the same way. This is annoying and unnecessary if the field with the categories remain visible.

**Describe the solution you'd like**
- implement the categories in the SliverAppBar with floating mode (FlexibleSpaceBar should do the trick)
- ensure that the categories will hide on scroll down to improve list size
- ensure that the categories will be shown on scroll up

[Design](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?t=FbETZJIKTLlAjzLy-0)

Further infos to Sliver App Bar including example videos:
https://api.flutter.dev/flutter/material/SliverAppBar-class.html

**Comment by f1sh1918** [comment]
@hauf-toni at the moment there is a "In meiner Nähe suchen" floating action button, so i don't know if its a good ui/ux to put another button next to it.
Maybe you can add a proposal for a design

**Comment by seluianova** [comment]
As another alternative idea, maybe clicking "Suche" again on the bottom bar should refresh the page and thus return to the top?

**Comment by hauf-toni** [comment]
You can find my design suggestion in this [📍 Figma file](https://www.figma.com/file/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?type=design&node-id=522%3A43&mode=design&t=I6zFZXMNB90PFhm3-1)

**Comment by steffenkleinle** [comment]
Is this still in design review?

**Comment by hauf-toni** [comment]
@f1sh1918 can you give us an insight into the current status?

**Comment by f1sh1918** [comment]
@hauf-toni 
I like the filter for categories button.
i'm not pretty sure about the flow.
When will this button appear instead of the Categories list?
Directly when the user begins to scroll? Or when a certain amount of items were scrolled (which can be complicated)
Even it may look strange if you just scroll a little bit and the list directly transforms.
The same question in the other way. Will this button turn back to categories list and when?
It can be hard to indicate because were are using different smartphone sizes and amount of list items per view to make a good indication, so the behavior could be different depending on device size

**Comment by hauf-toni** [comment]
@f1sh1918 thank you for your questions, here are my thoughts on this:

- when will the button appear? the threshold should be set based on a percentage of the screen height rather than a fixed number of items: this adapts better to different screen sizes, eg once at least 1-2 full rows of results are out of view
- the categories should reappear on scroll-up: the filter button could remain as long as the user scrolls downward; fi the user scrolls back up, the full category list could reappear (when reaching a similar threshold in the other direction)
- we should aim for a seamless flow guided by visual aid and implement a smooth fade-out/collapse animation/etc. for the categories

what do you think? let me know if there any questions that remain unclear

**Comment by seluianova** [comment]
looks good to me in general.

just got one more idea to share, feel free to ignore though:
what if we put the "filter" icon here?
so that when you click on it, the filter section will collapse/expand.
(the current "search" icon does nothing)
![Image](https://github.com/user-attachments/assets/22d6b34f-4231-484b-a7a4-36562c8ccdc3)

**Comment by hauf-toni** [comment]
@seluianova I like it! I will link an updated design proposal in this thread next week.

**Comment by hauf-toni** [comment]
@connium and me just discussed this issue. an updated proposal can be found [📌 here](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?node-id=1024-311&t=Mtx0I29ZJauY4bUa-1). @f1sh1918 and maybe @steffenkleinle could you review this proposal and let me know your feedback? thanks!

**Comment by seluianova** [comment]
> The transition from the filter list to the button should happen immedialtely on downscroll. when users scroll back up, the list should reappear. This way, the user experience remains seamless and intuitive.

Maybe in that case we don't even need an additional button to collapse/expand the filter?

**Comment by hauf-toni** [comment]
> > The transition from the filter list to the button should happen immedialtely on downscroll. when users scroll back up, the list should reappear. This way, the user experience remains seamless and intuitive.
> 
> Maybe in that case we don't even need an additional button to collapse/expand the filter?

You're right – see "design" [here](https://www.figma.com/design/SDoGnXIjCCXpE1tAAJkRVc/entitlementcard?node-id=1134-1200&t=maAt5BnShuOdnL0s-1)!

**Comment by seluianova** [comment]
I like the new design 👍😀

**Comment by f1sh1918** [comment]
@hauf-toni the issue can be moved to done for ui/ux

Thx :)
